indexinglanguage/src/main/java/com/yahoo/vespa/indexinglanguage/linguistics/LinguisticsAnnotator.java

```diff
@@ -116,42 +116,39 @@ private static void addAnnotationSpan(String input, SpanList parent, Token token
             }
             if ( ! token.isIndexable()) return;
         }
-        String orig = token.getOrig();
-        int pos = (int)token.getOffset();
-        if (pos >= input.length()) {
-            throw new IllegalArgumentException("Token '" + orig + "' has offset " + pos + ", which is outside the " +
-                                               "bounds of the input string; " + input);
+        if (token.getOffset() >= input.length()) {
```

Comment:
This does not come without a performance cost.
At least one of the Token implementations uses lazy lookup of the token when calling getOrig(),
and getOffset will cause a non cached jni call that we want to avoid.